{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os, json\n",
    "import nltk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATAPATH = 'data'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_filepath =\"/\".join([DATAPATH, 'metadata.csv'])\n",
    "df = pd.read_csv(csv_filepath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_pdf_parse</th>\n",
       "      <th>has_pmc_xml_parse</th>\n",
       "      <th>full_text_file</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>xqhn0vbp</td>\n",
       "      <td>1e1286db212100993d03cc22374b624f7caee956</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Airborne rhinovirus detection and effect of ul...</td>\n",
       "      <td>10.1186/1471-2458-3-5</td>\n",
       "      <td>PMC140314</td>\n",
       "      <td>12525263.0</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>BACKGROUND: Rhinovirus, the most common cause ...</td>\n",
       "      <td>2003-01-13</td>\n",
       "      <td>Myatt, Theodore A; Johnston, Sebastian L; Rudn...</td>\n",
       "      <td>BMC Public Health</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gi6uaa83</td>\n",
       "      <td>8ae137c8da1607b3a8e4c946c07ca8bda67f88ac</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Discovering human history from stomach bacteria</td>\n",
       "      <td>10.1186/gb-2003-4-5-213</td>\n",
       "      <td>PMC156578</td>\n",
       "      <td>12734001.0</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Recent analyses of human pathogens have reveal...</td>\n",
       "      <td>2003-04-28</td>\n",
       "      <td>Disotell, Todd R</td>\n",
       "      <td>Genome Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>le0ogx1s</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>A new recruit for the army of the men of death</td>\n",
       "      <td>10.1186/gb-2003-4-7-113</td>\n",
       "      <td>PMC193621</td>\n",
       "      <td>12844350.0</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>The army of the men of death, in John Bunyan's...</td>\n",
       "      <td>2003-06-27</td>\n",
       "      <td>Petsko, Gregory A</td>\n",
       "      <td>Genome Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fy4w7xz8</td>\n",
       "      <td>0104f6ceccf92ae8567a0102f89cbb976969a774</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Association of HLA class I with severe acute r...</td>\n",
       "      <td>10.1186/1471-2350-4-9</td>\n",
       "      <td>PMC212558</td>\n",
       "      <td>12969506.0</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>BACKGROUND: The human leukocyte antigen (HLA) ...</td>\n",
       "      <td>2003-09-12</td>\n",
       "      <td>Lin, Marie; Tseng, Hsiang-Kuang; Trejaut, Jean...</td>\n",
       "      <td>BMC Med Genet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0qaoam29</td>\n",
       "      <td>5b68a553a7cbbea13472721cd1ad617d42b40c26</td>\n",
       "      <td>PMC</td>\n",
       "      <td>A double epidemic model for the SARS propagation</td>\n",
       "      <td>10.1186/1471-2334-3-19</td>\n",
       "      <td>PMC222908</td>\n",
       "      <td>12964944.0</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>BACKGROUND: An epidemic of a Severe Acute Resp...</td>\n",
       "      <td>2003-09-10</td>\n",
       "      <td>Ng, Tuen Wai; Turinici, Gabriel; Danchin, Antoine</td>\n",
       "      <td>BMC Infect Dis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   cord_uid                                       sha source_x  \\\n",
       "0  xqhn0vbp  1e1286db212100993d03cc22374b624f7caee956      PMC   \n",
       "1  gi6uaa83  8ae137c8da1607b3a8e4c946c07ca8bda67f88ac      PMC   \n",
       "2  le0ogx1s                                       NaN      PMC   \n",
       "3  fy4w7xz8  0104f6ceccf92ae8567a0102f89cbb976969a774      PMC   \n",
       "4  0qaoam29  5b68a553a7cbbea13472721cd1ad617d42b40c26      PMC   \n",
       "\n",
       "                                               title                      doi  \\\n",
       "0  Airborne rhinovirus detection and effect of ul...    10.1186/1471-2458-3-5   \n",
       "1    Discovering human history from stomach bacteria  10.1186/gb-2003-4-5-213   \n",
       "2     A new recruit for the army of the men of death  10.1186/gb-2003-4-7-113   \n",
       "3  Association of HLA class I with severe acute r...    10.1186/1471-2350-4-9   \n",
       "4   A double epidemic model for the SARS propagation   10.1186/1471-2334-3-19   \n",
       "\n",
       "       pmcid   pubmed_id license  \\\n",
       "0  PMC140314  12525263.0   no-cc   \n",
       "1  PMC156578  12734001.0   no-cc   \n",
       "2  PMC193621  12844350.0   no-cc   \n",
       "3  PMC212558  12969506.0   no-cc   \n",
       "4  PMC222908  12964944.0   no-cc   \n",
       "\n",
       "                                            abstract publish_time  \\\n",
       "0  BACKGROUND: Rhinovirus, the most common cause ...   2003-01-13   \n",
       "1  Recent analyses of human pathogens have reveal...   2003-04-28   \n",
       "2  The army of the men of death, in John Bunyan's...   2003-06-27   \n",
       "3  BACKGROUND: The human leukocyte antigen (HLA) ...   2003-09-12   \n",
       "4  BACKGROUND: An epidemic of a Severe Acute Resp...   2003-09-10   \n",
       "\n",
       "                                             authors            journal  \\\n",
       "0  Myatt, Theodore A; Johnston, Sebastian L; Rudn...  BMC Public Health   \n",
       "1                                   Disotell, Todd R        Genome Biol   \n",
       "2                                  Petsko, Gregory A        Genome Biol   \n",
       "3  Lin, Marie; Tseng, Hsiang-Kuang; Trejaut, Jean...      BMC Med Genet   \n",
       "4  Ng, Tuen Wai; Turinici, Gabriel; Danchin, Antoine     BMC Infect Dis   \n",
       "\n",
       "   Microsoft Academic Paper ID WHO #Covidence  has_pdf_parse  \\\n",
       "0                          NaN            NaN           True   \n",
       "1                          NaN            NaN           True   \n",
       "2                          NaN            NaN          False   \n",
       "3                          NaN            NaN           True   \n",
       "4                          NaN            NaN           True   \n",
       "\n",
       "   has_pmc_xml_parse  full_text_file  \\\n",
       "0               True  custom_license   \n",
       "1               True  custom_license   \n",
       "2               True  custom_license   \n",
       "3               True  custom_license   \n",
       "4               True  custom_license   \n",
       "\n",
       "                                                 url  \n",
       "0  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...  \n",
       "1  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...  \n",
       "2  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...  \n",
       "3  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...  \n",
       "4  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(42511, 18)"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sub = df[df['full_text_file'].notna() == True]\n",
    "df_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(35074, 18)"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sub = df_sub[df_sub['abstract'].notna() == True]\n",
    "df_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(34737, 18)"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sub = df_sub[df_sub['abstract'] != 'Unknown']\n",
    "df_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(33878, 18)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sub = df_sub[(df_sub['has_pdf_parse'] == True) | (df_sub['has_pmc_xml_parse'] == True)]\n",
    "df_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Khachatur\\Miniconda3\\envs\\tensorflow\\lib\\site-packages\\ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(33526, 18)"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sub = df_sub[df['abstract'].str.len() >= 250]\n",
    "df_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "  \"paper_id\": \"8ae137c8da1607b3a8e4c946c07ca8bda67f88ac\",\n",
      "  \"metadata\": {\n",
      "    \"title\": \"Discovering human history from stomach bacteria\",\n",
      "    \"authors\": [\n",
      "      {\n",
      "        \"first\": \"Todd\",\n",
      "        \"middle\": [\n",
      "          \"R\"\n",
      "        ],\n",
      "        \"last\": \"Disotell\",\n",
      "        \"suffix\": \"\",\n",
      "        \"affiliation\": {\n",
      "          \"laboratory\": \"\",\n",
      "          \"institution\": \"New York University\",\n",
      "          \"location\": {\n",
      "            \"addrLine\": \"25 Waverly Place\",\n",
      "            \"postCode\": \"10003\",\n",
      "            \"settlement\": \"New York\",\n",
      "            \"region\": \"NY\",\n",
      "            \"country\": \"USA\"\n",
      "          }\n",
      "        },\n",
      "        \"email\": \"todd.disotell@nyu.edu\"\n",
      "      }\n",
      "    ]\n",
      "  },\n",
      "  \"abstract\": [\n",
      "    {\n",
      "      \"text\": \"comment reviews reports deposited research interactions information refereed research Abstract Recent analyses of human pathogens have revealed that their evolutionary histories are congruent with the hypothesized pattern of ancient and modern human population migrations.\",\n",
      "      \"cite_spans\": [],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"Abstract\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Phylogenetic trees of strains of the bacterium Helicobacter pylori and the polyoma JC virus taken from geographically diverse groups of human beings correlate closely with relationships of the populations in which they are found.\",\n",
      "      \"cite_spans\": [],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"Abstract\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"The electronic version of this article is the complete one and can be found online at\",\n",
      "      \"cite_spans\": [],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"Abstract\"\n",
      "    }\n",
      "  ],\n",
      "  \"body_text\": [\n",
      "    {\n",
      "      \"text\": \"Charles Darwin recognized that the distribution and form of parasites was evolutionarily significant. He noted, for instance, that \\\"... the Pediculi [lice] collected in different countries from the different races of man ... differ, not only in colour, but in the structure of their claws and limbs. In every case in which many specimens were obtained the differences were constant\\\" [1] . More recently, several research groups [2] [3] [4] [5] [6] [7] have found interesting correlations between the evolutionary relationships among various bacterial and viral strains hosted by humans and the pattern of migrations of modern humans throughout the world.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 383,\n",
      "          \"end\": 386,\n",
      "          \"text\": \"[1]\",\n",
      "          \"ref_id\": \"BIBREF0\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 428,\n",
      "          \"end\": 431,\n",
      "          \"text\": \"[2]\",\n",
      "          \"ref_id\": \"BIBREF1\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 432,\n",
      "          \"end\": 435,\n",
      "          \"text\": \"[3]\",\n",
      "          \"ref_id\": \"BIBREF2\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 436,\n",
      "          \"end\": 439,\n",
      "          \"text\": \"[4]\",\n",
      "          \"ref_id\": \"BIBREF3\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 440,\n",
      "          \"end\": 443,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 444,\n",
      "          \"end\": 447,\n",
      "          \"text\": \"[6]\",\n",
      "          \"ref_id\": \"BIBREF5\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 448,\n",
      "          \"end\": 451,\n",
      "          \"text\": \"[7]\",\n",
      "          \"ref_id\": \"BIBREF6\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"A particularly interesting case is that of Helicobacter pylori, a Gram-negative bacterium associated with gastritis, peptic ulcers, and gastric cancer that may infect up to half of all humans [8] . The discovery that a bacterial infection could lead to what were considered chronic diseases [8] was a striking example of the fact that infectious diseases have not yet been conquered. The continuing acquired immune deficiency syndrome (AIDS) epidemic, outbreaks of Ebola in Central Africa, and the current spread of West Nile Virus in the United States and of severe acute respiratory syndrome (SARS) from Asia provide evidence of the pervasiveness and health consequences of infectious agents even in the age of vaccination and antimicrobial and antiviral therapies. Many infectious diseases are thought to have arisen concurrently with the development of agriculture and the rise of urban living. If, instead, many pathogens' relationships with humans are much older, it would not be surprising to find deeper evolutionary associations between humans and their microbial and viral invaders.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 192,\n",
      "          \"end\": 195,\n",
      "          \"text\": \"[8]\",\n",
      "          \"ref_id\": \"BIBREF7\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 291,\n",
      "          \"end\": 294,\n",
      "          \"text\": \"[8]\",\n",
      "          \"ref_id\": \"BIBREF7\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"The evolutionary history of H. pylori may provide an example of the coevolution of a bacterium and its only known host. The H. pylori genome is relatively small at 1.67 megabases, with a minimal complement of metabolic genes [9] . Variation between H. pylori isolates from different people or even from one person is large, leading to unique fingerprints for nearly every isolate so far typed. The coding genes are not very diverse, however: most of the variation occurs in the third base position within codons or through inversions or translocations, leaving the encoded amino-acid sequences relatively similar [2] . This amino-acid sequence conservation is fortunate for vaccine research, as one vaccine is likely to be effective on many strains. More interesting is the fact that H. pylori has an extremely high rate of recombination, higher in fact than that of any other organism characterized to date [3, 10] .\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 225,\n",
      "          \"end\": 228,\n",
      "          \"text\": \"[9]\",\n",
      "          \"ref_id\": \"BIBREF8\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 613,\n",
      "          \"end\": 616,\n",
      "          \"text\": \"[2]\",\n",
      "          \"ref_id\": \"BIBREF1\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 908,\n",
      "          \"end\": 911,\n",
      "          \"text\": \"[3,\",\n",
      "          \"ref_id\": \"BIBREF2\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 912,\n",
      "          \"end\": 915,\n",
      "          \"text\": \"10]\",\n",
      "          \"ref_id\": \"BIBREF9\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Normally, such a high rate of recombination would make inferring an organism's evolutionary history very difficult, as information about the origin of each mutation would be lost as it spreads throughout a population. But coupled with the mode of transmission of H. pylori, this extremely high rate of recombination may in fact make evolutionary inferences easier. Several studies strongly suggest that H. pylori is usually transmitted within families, generally from mother to child [11, 12] . Thus, the transmission of H. pylori in some ways mimics that of maternally transmitted mitochondrial DNA [13] . Because mitochondrial DNA is transmitted solely from one parent (the mother) and does not recombine, it has proved to be an ideal genetic system for inferring human evolutionary history (see below) [14, 15] . If H. pylori is indeed predominantly maternally transmitted, new strains will generally not infect a person during their lifetime; together with the high rate of recombination, this would mean that the mutations that accumulate within the population of bacteria in an individual's stomach will be relatively homogeneous. This should result in a swarm of strains that are very closely related to each other, containing many of the mutations that have occurred in individual bacteria. Swarms found in different people will thus be more different from each other than if there was less recombination.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 484,\n",
      "          \"end\": 488,\n",
      "          \"text\": \"[11,\",\n",
      "          \"ref_id\": \"BIBREF10\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 489,\n",
      "          \"end\": 492,\n",
      "          \"text\": \"12]\",\n",
      "          \"ref_id\": \"BIBREF11\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 600,\n",
      "          \"end\": 604,\n",
      "          \"text\": \"[13]\",\n",
      "          \"ref_id\": \"BIBREF12\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 805,\n",
      "          \"end\": 809,\n",
      "          \"text\": \"[14,\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 810,\n",
      "          \"end\": 813,\n",
      "          \"text\": \"15]\",\n",
      "          \"ref_id\": \"BIBREF14\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Most infectious diseases spread quickly throughout the world and strains from different regions are relatively similar, but initial sampling of H. pylori from people from different regions of the world revealed fairly strong geographic partitioning into European and Asian H. pylori types [2] [3] [4] . Recently, Falush and colleagues [5] have examined this partitioning in greater detail. After sequencing eight genes -a total of 3,850 nucleotides -in 370 strains derived from 27 human populations, they found 1,418 polymorphic nucleotide positions. They then applied a new analytic tool, STRUCTURE [16] , that was developed to infer human genetic structure from multilocus genotype data. The program uses Bayesian methods to identify subgroups with distinctive allele frequencies, and clusters of the subgroups, even in the presence of recombination [16] . When this technique was applied to the H. pylori sequences, four main clusters were found -two from Africa and one each from Europe and Asia (Figure 1a ) [5] .\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 289,\n",
      "          \"end\": 292,\n",
      "          \"text\": \"[2]\",\n",
      "          \"ref_id\": \"BIBREF1\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 293,\n",
      "          \"end\": 296,\n",
      "          \"text\": \"[3]\",\n",
      "          \"ref_id\": \"BIBREF2\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 297,\n",
      "          \"end\": 300,\n",
      "          \"text\": \"[4]\",\n",
      "          \"ref_id\": \"BIBREF3\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 335,\n",
      "          \"end\": 338,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 600,\n",
      "          \"end\": 604,\n",
      "          \"text\": \"[16]\",\n",
      "          \"ref_id\": \"BIBREF15\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 852,\n",
      "          \"end\": 856,\n",
      "          \"text\": \"[16]\",\n",
      "          \"ref_id\": \"BIBREF15\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 1013,\n",
      "          \"end\": 1016,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [\n",
      "        {\n",
      "          \"start\": 1000,\n",
      "          \"end\": 1010,\n",
      "          \"text\": \"(Figure 1a\",\n",
      "          \"ref_id\": null\n",
      "        }\n",
      "      ],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Each cluster found by Falush and colleagues [5] could be divided into subgroups; for instance, the 'Africa 1' cluster could be subdivided further into West and South African subclusters, and the East Asia cluster could be split into East Asian, Amerind, and Maori subclusters. The geographic partitioning within the 200 European strains was particularly complicated, presumably because numerous groups have swept back and forth across Europe over the past several millennia. European strains also occasionally appeared in the Americas, Australia, and among South Africans, presumably reflecting colonial conquest.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 44,\n",
      "          \"end\": 47,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"The phylogenetic relationships of these clusters (Figure 1b) and their subdivisions [5] show a pattern similar to that The relationships between human populations, as calculated from H. pylori found in stomachs and from mitochondrial DNA data. (a) Relationships between modern subpopulations of H. pylori [5] . Each subpopulation is represented by a circle with a diameter proportional to the genetic diversity within it. The centres of the circles are joined by a phylogenetic tree showing the relationships between the four subpopulations. Bacteria in each subpopulation are found predominantly in people who originate from the regions shown. (b) A population-level phylogenetic tree of the H. pylori geographic subpopulations shown in (a). (c) A median-joining network of human populations derived from mitochondrial DNA [14] . Such a network shows alternative potential evolutionary relationships between clusters. Each circle represents a cluster of mitochondrial types with a diameter proportional to the frequency of that type within the subpopulations. All non-African populations are derived from one African lineage; the network of relationships within this lineage is magnified (top). (a,b) Adapted from [5] ; (c) adapted from [14] . \",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 84,\n",
      "          \"end\": 87,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 305,\n",
      "          \"end\": 308,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 824,\n",
      "          \"end\": 828,\n",
      "          \"text\": \"[14]\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 1215,\n",
      "          \"end\": 1218,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 1238,\n",
      "          \"end\": 1242,\n",
      "          \"text\": \"[14]\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [\n",
      "        {\n",
      "          \"start\": 49,\n",
      "          \"end\": 60,\n",
      "          \"text\": \"(Figure 1b)\",\n",
      "          \"ref_id\": null\n",
      "        }\n",
      "      ],\n",
      "      \"section\": \"\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"obtained using mitochondrial DNA variation (Figure 1c ) [14, 15] . The modern human gene pool, as inferred from mitochondrial DNA and corroborated by Y chromosome studies, is thought to have had an African origin approximately 150,000-200,000 years ago [14, 15, 17, 18] . The original human population then spread and diversified throughout Africa for nearly 100,000 years, before expanding into Western Asia and Europe and into Southern and Eastern Asia approximately 50,000-60,000 years ago, replacing the existing archaic populations of humans in these regions. Subsequent migrations spread into Australasia by 40,000 years ago, then to the Pacific Islands, and later into North America, approximately 15,000 years ago ( Figure 2 ) [14, 15, 17, 18] . The remarkable similarity between this view of human history and the results from studies of H. pylori have led Falush and colleagues [5] as well as others [2] to conclude that H. pylori evolution has followed the path of modern human expansion and migration. This work thus provides another type of data for analysis of human evolution and migration, independent of mitochondrial DNA and Y chromosomes, that will be valuable in further studies. Unfortunately, however, estimating the divergence dates from H. pylori is particularly difficult, owing to the extremely high rate of recombination [10] . Additional sampling and analytical techniques may be required to further test the migratory hypothesis.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 56,\n",
      "          \"end\": 60,\n",
      "          \"text\": \"[14,\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 61,\n",
      "          \"end\": 64,\n",
      "          \"text\": \"15]\",\n",
      "          \"ref_id\": \"BIBREF14\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 253,\n",
      "          \"end\": 257,\n",
      "          \"text\": \"[14,\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 258,\n",
      "          \"end\": 261,\n",
      "          \"text\": \"15,\",\n",
      "          \"ref_id\": \"BIBREF14\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 262,\n",
      "          \"end\": 265,\n",
      "          \"text\": \"17,\",\n",
      "          \"ref_id\": \"BIBREF16\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 266,\n",
      "          \"end\": 269,\n",
      "          \"text\": \"18]\",\n",
      "          \"ref_id\": \"BIBREF17\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 735,\n",
      "          \"end\": 739,\n",
      "          \"text\": \"[14,\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 740,\n",
      "          \"end\": 743,\n",
      "          \"text\": \"15,\",\n",
      "          \"ref_id\": \"BIBREF14\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 744,\n",
      "          \"end\": 747,\n",
      "          \"text\": \"17,\",\n",
      "          \"ref_id\": \"BIBREF16\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 748,\n",
      "          \"end\": 751,\n",
      "          \"text\": \"18]\",\n",
      "          \"ref_id\": \"BIBREF17\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 888,\n",
      "          \"end\": 891,\n",
      "          \"text\": \"[5]\",\n",
      "          \"ref_id\": \"BIBREF4\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 910,\n",
      "          \"end\": 913,\n",
      "          \"text\": \"[2]\",\n",
      "          \"ref_id\": \"BIBREF1\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 1348,\n",
      "          \"end\": 1352,\n",
      "          \"text\": \"[10]\",\n",
      "          \"ref_id\": \"BIBREF9\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [\n",
      "        {\n",
      "          \"start\": 43,\n",
      "          \"end\": 53,\n",
      "          \"text\": \"(Figure 1c\",\n",
      "          \"ref_id\": null\n",
      "        },\n",
      "        {\n",
      "          \"start\": 724,\n",
      "          \"end\": 732,\n",
      "          \"text\": \"Figure 2\",\n",
      "          \"ref_id\": null\n",
      "        }\n",
      "      ],\n",
      "      \"section\": \"(a) (b) (c)\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Other pathogens have also been proposed to follow evolutionary histories similar to their hosts. One of the most interesting examples, though with a non-human host, is that of aphids, a bacterium found within them, and two plasmids associated with the bacterium. Funk et al. [19] found that the inferred intraspecific phylogenies of these four genomes were completely congruent. Returning to human pathogens, the human polyomavirus JC virus (JCV) can be divided into genotypes that correspond to the major continental land masses [6] . Like H. pylori, JCV -which can cause progressive multifocal leukoencephalopathy (loss of myelination in the central nervous system) -is very widespread amongst humans as a result of familial transmission. A total of 12 known subtypes have been defined, with European, African, and Asian distributions [7] . Although direct inferences of an African origin for JCV are problematic because there is no suitable outgroup with which to root the phylogenetic tree, when an African origin is assumed, a reasonable evolutionary history can be hypothesized ( Figure 3 ) [7] . As with H. pylori, inferring molecular divergence dates is currently problematic for JCV. Further investigation into the evolutionary history of these human pathogens is therefore necessary.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 275,\n",
      "          \"end\": 279,\n",
      "          \"text\": \"[19]\",\n",
      "          \"ref_id\": \"BIBREF18\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 530,\n",
      "          \"end\": 533,\n",
      "          \"text\": \"[6]\",\n",
      "          \"ref_id\": \"BIBREF5\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 837,\n",
      "          \"end\": 840,\n",
      "          \"text\": \"[7]\",\n",
      "          \"ref_id\": \"BIBREF6\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 1097,\n",
      "          \"end\": 1100,\n",
      "          \"text\": \"[7]\",\n",
      "          \"ref_id\": \"BIBREF6\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [\n",
      "        {\n",
      "          \"start\": 1086,\n",
      "          \"end\": 1094,\n",
      "          \"text\": \"Figure 3\",\n",
      "          \"ref_id\": null\n",
      "        }\n",
      "      ],\n",
      "      \"section\": \"(a) (b) (c)\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Elucidating the patterns of evolution of human pathogens may ultimately provide additional evidence not only about their history but also about human evolution and history. This will be especially true for pathogens such as H. pylori that have a predominantly mother-child mode of transmission, mimicking mitochondrial DNA evolution. H. pylori's causative role in several chronic stomach conditions is just one of many current examples of the known or suspected \",\n",
      "      \"cite_spans\": [],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"(a) (b) (c)\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"A map of the pattern of expansion and migration of modern humans throughout the world, derived from studies of mitochondrial DNA and Y chromosomes [14, 15, 17, 18] . Numbers indicate the approximate time (in years before the present) when modern humans first appeared in the indicated region. role of an infectious agent leading to chronic diseases. Bacteria are suspected to be involved in the development of arteriosclerosis, stroke, and Crohn's disease, whereas viruses are known to lead to AIDS and the various forms of chronic hepatitis. Cervical cancer, hepatocellular carcinoma, Burkitt's lymphoma, Kaposi's sarcoma, and perhaps diabetes mellitus are also either known or suspected to be of viral origin. If some of the ubiquitous chronic diseases prove to be of bacterial or viral origin, worldwide surveys of these pathogens in human populations should be carried out immediately, so that knowledge about the evolution and diversity of these pathogens can be incorporated into the research programs designed to ameliorate the conditions that they cause.\",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 147,\n",
      "          \"end\": 151,\n",
      "          \"text\": \"[14,\",\n",
      "          \"ref_id\": \"BIBREF13\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 152,\n",
      "          \"end\": 155,\n",
      "          \"text\": \"15,\",\n",
      "          \"ref_id\": \"BIBREF14\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 156,\n",
      "          \"end\": 159,\n",
      "          \"text\": \"17,\",\n",
      "          \"ref_id\": \"BIBREF16\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 160,\n",
      "          \"end\": 163,\n",
      "          \"text\": \"18]\",\n",
      "          \"ref_id\": \"BIBREF17\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"Figure 2\"\n",
      "    },\n",
      "    {\n",
      "      \"text\": \"Relationships of human polyoma JC virus (JCV) subtypes found in humans from different parts of the world [7] . Letters refer to individual subtypes. (a) The hypothesized pattern of spread of JCV subtypes through the world (excluding the Americas); (b) an inferred phylogeny of JCV subtypes, assuming an African origin for the virus. Adapted from [7] . \",\n",
      "      \"cite_spans\": [\n",
      "        {\n",
      "          \"start\": 105,\n",
      "          \"end\": 108,\n",
      "          \"text\": \"[7]\",\n",
      "          \"ref_id\": \"BIBREF6\"\n",
      "        },\n",
      "        {\n",
      "          \"start\": 346,\n",
      "          \"end\": 349,\n",
      "          \"text\": \"[7]\",\n",
      "          \"ref_id\": \"BIBREF6\"\n",
      "        }\n",
      "      ],\n",
      "      \"ref_spans\": [],\n",
      "      \"section\": \"Figure 3\"\n",
      "    }\n",
      "  ],\n",
      "  \"bib_entries\": {\n",
      "    \"BIBREF0\": {\n",
      "      \"ref_id\": \"b0\",\n",
      "      \"title\": \"The Descent of Man and Selection in Relation to Sex\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"C\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Darwin\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": null,\n",
      "      \"venue\": \"\",\n",
      "      \"volume\": \"\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF1\": {\n",
      "      \"ref_id\": \"b1\",\n",
      "      \"title\": \"Helicobacter pylori virulence and genetic geography\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Covacci\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"L\"\n",
      "          ],\n",
      "          \"last\": \"Telford\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"Del\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Giudice\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Parsonnet\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Rappuoli\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"R\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1999,\n",
      "      \"venue\": \"Science\",\n",
      "      \"volume\": \"284\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"1328--1333\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF2\": {\n",
      "      \"ref_id\": \"b2\",\n",
      "      \"title\": \"Evolution of Helicobacter pylori: the role of recombination\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Suerbaum\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Achtman\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1999,\n",
      "      \"venue\": \"Trends Microbiol\",\n",
      "      \"volume\": \"7\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"182--184\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF3\": {\n",
      "      \"ref_id\": \"b3\",\n",
      "      \"title\": \"Recombination and clonal groupings within Helicobacter pylori from different geographical regions\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Achtman\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"T\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Azuma\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [\n",
      "            \"E\"\n",
      "          ],\n",
      "          \"last\": \"Berg\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"Y\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Ito\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Morelli\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"Z-J\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Pan\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Suerbaum\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Thompson\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Van Der Ende\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"L-J\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Van Doorn\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1999,\n",
      "      \"venue\": \"Mol Microbiol\",\n",
      "      \"volume\": \"32\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"459--470\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF4\": {\n",
      "      \"ref_id\": \"b4\",\n",
      "      \"title\": \"Traces of human migrations in Helicobacter pylori populations\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Falush\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"T\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Wirth\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"B\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Linz\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"K\"\n",
      "          ],\n",
      "          \"last\": \"Pritchard\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Stephens\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kidd\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Blaser\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [\n",
      "            \"Y\"\n",
      "          ],\n",
      "          \"last\": \"Graham\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Vacher\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [\n",
      "            \"I\"\n",
      "          ],\n",
      "          \"last\": \"Perez-Perez\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2003,\n",
      "      \"venue\": \"\",\n",
      "      \"volume\": \"299\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"1582--1585\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF5\": {\n",
      "      \"ref_id\": \"b5\",\n",
      "      \"title\": \"Genotypes of JC virus in East, Central and Southwest Europe\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [\n",
      "            \"T\"\n",
      "          ],\n",
      "          \"last\": \"Agostini\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Deckhut\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [\n",
      "            \"V\"\n",
      "          ],\n",
      "          \"last\": \"Jobes\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"R\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Girones\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Schlunck\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [\n",
      "            \"G\"\n",
      "          ],\n",
      "          \"last\": \"Prost\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"C\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Frias\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"E\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"P\\u00e9rez-Trallero\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"C\",\n",
      "          \"middle\": [\n",
      "            \"F\"\n",
      "          ],\n",
      "          \"last\": \"Ryschkewitsch\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [\n",
      "            \"L\"\n",
      "          ],\n",
      "          \"last\": \"Stoner\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2001,\n",
      "      \"venue\": \"J Gen Virol\",\n",
      "      \"volume\": \"82\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"1221--1331\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF6\": {\n",
      "      \"ref_id\": \"b6\",\n",
      "      \"title\": \"Evolution of human polyomvirus JC: implications for the population history of humans\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"C\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Sugimoto\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Hasegawa\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kato\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H-Y\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Zheng\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Ebihara\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"F\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Taguchi\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"T\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kitamura\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"Y\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Yogo\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2002,\n",
      "      \"venue\": \"J Mol Evol\",\n",
      "      \"volume\": \"54\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"285--297\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF7\": {\n",
      "      \"ref_id\": \"b7\",\n",
      "      \"title\": \"Unidentified curved bacilli in the stomach of patients with gastritis and peptic ulceration\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"B\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Marshall\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"R\"\n",
      "          ],\n",
      "          \"last\": \"Warren\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1984,\n",
      "      \"venue\": \"Lancet\",\n",
      "      \"volume\": \"1\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"1311--1315\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF8\": {\n",
      "      \"ref_id\": \"b8\",\n",
      "      \"title\": \"The complete genome sequence of the gastric pathogen Helicobacter pylori\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"F\"\n",
      "          ],\n",
      "          \"last\": \"Tomb\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"O\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"White\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [\n",
      "            \"R\"\n",
      "          ],\n",
      "          \"last\": \"Kerlavage\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"R\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Clayton\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [\n",
      "            \"G\"\n",
      "          ],\n",
      "          \"last\": \"Sutton\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"R\",\n",
      "          \"middle\": [\n",
      "            \"D\"\n",
      "          ],\n",
      "          \"last\": \"Fleischmann\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"K\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Ketchum\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [\n",
      "            \"P\"\n",
      "          ],\n",
      "          \"last\": \"Klenk\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Gill\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"B\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Dougherty\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1997,\n",
      "      \"venue\": \"Nature\",\n",
      "      \"volume\": \"388\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"539--547\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF9\": {\n",
      "      \"ref_id\": \"b9\",\n",
      "      \"title\": \"Recombination and mutation during long-term gastric colonization by Helicobacter pylori: estimates of clock rates, recombination size, and minimal age\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Falush\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"C\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kraft\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"N\",\n",
      "          \"middle\": [\n",
      "            \"S\"\n",
      "          ],\n",
      "          \"last\": \"Taylor\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"P\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Correa\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"G\"\n",
      "          ],\n",
      "          \"last\": \"Fox\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Achtman\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Suerbaum\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2001,\n",
      "      \"venue\": \"Proc Natl Acad Sci\",\n",
      "      \"volume\": \"98\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"15056--15061\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF10\": {\n",
      "      \"ref_id\": \"b10\",\n",
      "      \"title\": \"Helicobacter pylori among preschool children and their parents: evidence of parent-child transmission\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Rothenbacher\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Bode\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Berg\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"U\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Knayer\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"T\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Gonser\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Adler\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Brenner\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1999,\n",
      "      \"venue\": \"J Infect Dis\",\n",
      "      \"volume\": \"179\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"398--402\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF11\": {\n",
      "      \"ref_id\": \"b11\",\n",
      "      \"title\": \"Helicobacter pylori infection in Swedish school children: lack of evidence of child-to-child transmission outside the family\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"Y\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Tindberg\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"C\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Bengtsson\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"F\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Granath\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Blennow\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"O\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Nyr\\u00e9n\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Granstr\\u00f6m\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2001,\n",
      "      \"venue\": \"Gastroenterology\",\n",
      "      \"volume\": \"121\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"310--316\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF12\": {\n",
      "      \"ref_id\": \"b12\",\n",
      "      \"title\": \"Maternal inheritance of human mitochondrial DNA\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"R\",\n",
      "          \"middle\": [\n",
      "            \"E\"\n",
      "          ],\n",
      "          \"last\": \"Giles\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Blanc\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [\n",
      "            \"M\"\n",
      "          ],\n",
      "          \"last\": \"Cann\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [\n",
      "            \"C\"\n",
      "          ],\n",
      "          \"last\": \"Wallace\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1980,\n",
      "      \"venue\": \"Proc Natl Acad Sci\",\n",
      "      \"volume\": \"77\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"6715--6719\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF13\": {\n",
      "      \"ref_id\": \"b13\",\n",
      "      \"title\": \"Deep common ancestry of Indian and western Eurasian mtDNA lineages\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"T\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kivisild\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Bamshad\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"K\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kaldma\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Metspalu\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"E\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Metspalu\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Reidla\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Laos\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Parik\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"W\",\n",
      "          \"middle\": [\n",
      "            \"S\"\n",
      "          ],\n",
      "          \"last\": \"Watkins\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [\n",
      "            \"E\"\n",
      "          ],\n",
      "          \"last\": \"Dixon\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 1999,\n",
      "      \"venue\": \"Curr Biol\",\n",
      "      \"volume\": \"9\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"1331--1334\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF14\": {\n",
      "      \"ref_id\": \"b14\",\n",
      "      \"title\": \"Mitochondrial genome variation and the origin of modern humans\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Ingman\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Kaessmann\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Paabo\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"U\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Gyllensten\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2000,\n",
      "      \"venue\": \"Nature\",\n",
      "      \"volume\": \"408\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"708--713\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF15\": {\n",
      "      \"ref_id\": \"b15\",\n",
      "      \"title\": \"Inference of population structure using multilocus genotype data\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"K\"\n",
      "          ],\n",
      "          \"last\": \"Pritchard\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Stephens\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"P\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Donnelly\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2000,\n",
      "      \"venue\": \"Genetics\",\n",
      "      \"volume\": \"155\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"945--959\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF16\": {\n",
      "      \"ref_id\": \"b16\",\n",
      "      \"title\": \"Hierarchical patterns of global human Y-chromosome diversity\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [\n",
      "            \"F\"\n",
      "          ],\n",
      "          \"last\": \"Hammer\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"T\",\n",
      "          \"middle\": [\n",
      "            \"M\"\n",
      "          ],\n",
      "          \"last\": \"Karafet\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Redd\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Jarjanazi\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Santachiara-Benerecetti\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"H\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Soodyall\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"S\",\n",
      "          \"middle\": [\n",
      "            \"L\"\n",
      "          ],\n",
      "          \"last\": \"Zegura\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2001,\n",
      "      \"venue\": \"Mol Biol Evol\",\n",
      "      \"volume\": \"18\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"1189--1203\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF17\": {\n",
      "      \"ref_id\": \"b17\",\n",
      "      \"title\": \"The phylogeography of Y chromosome binary haplotypes and the origins of modern human populations\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"P\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Underhill\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"G\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Passarino\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"A\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Lin\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"P\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Shen\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"Mirazon\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Lahr\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"M\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Foley\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"R\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Oefner\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"P\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Cavalli-Sforza\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"L\",\n",
      "          \"middle\": [\n",
      "            \"L\"\n",
      "          ],\n",
      "          \"last\": \"\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2001,\n",
      "      \"venue\": \"Ann Hum Genet\",\n",
      "      \"volume\": \"65\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"43--62\",\n",
      "      \"other_ids\": {}\n",
      "    },\n",
      "    \"BIBREF18\": {\n",
      "      \"ref_id\": \"b18\",\n",
      "      \"title\": \"Intraspecific phylogenetic congruence among multiple symbiont genomes\",\n",
      "      \"authors\": [\n",
      "        {\n",
      "          \"first\": \"D\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Funk\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"L\",\n",
      "          \"middle\": [],\n",
      "          \"last\": \"Helbling\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"J\",\n",
      "          \"middle\": [\n",
      "            \"J\"\n",
      "          ],\n",
      "          \"last\": \"Wernegreen\",\n",
      "          \"suffix\": \"\"\n",
      "        },\n",
      "        {\n",
      "          \"first\": \"N\",\n",
      "          \"middle\": [\n",
      "            \"A\"\n",
      "          ],\n",
      "          \"last\": \"Moran\",\n",
      "          \"suffix\": \"\"\n",
      "        }\n",
      "      ],\n",
      "      \"year\": 2000,\n",
      "      \"venue\": \"Proc R Soc Lond B Biol Sci\",\n",
      "      \"volume\": \"267\",\n",
      "      \"issn\": \"\",\n",
      "      \"pages\": \"2517--2521\",\n",
      "      \"other_ids\": {}\n",
      "    }\n",
      "  },\n",
      "  \"ref_entries\": {\n",
      "    \"FIGREF0\": {\n",
      "      \"text\": \"Figure 1 The relationships between human populations, as calculated from H. pylori found in stomachs and from mitochondrial DNA data. (a) Relationships between modern subpopulations of H. pylori [5]. Each subpopulation is represented by a circle with a diameter proportional to the genetic diversity within it. The centres of the circles are joined by a phylogenetic tree showing the relationships between the four subpopulations. Bacteria in each subpopulation are found predominantly in people who originate from the regions shown. (b) A population-level phylogenetic tree of the H. pylori geographic subpopulations shown in (a). (c) A median-joining network of human populations derived from mitochondrial DNA [14]. Such a network shows alternative potential evolutionary relationships between clusters. Each circle represents a cluster of mitochondrial types with a diameter proportional to the frequency of that type within the subpopulations. All non-African populations are derived from one African lineage; the network of relationships within this lineage is magnified (top). (a,b) Adapted from [5]; (c) adapted from [14].\",\n",
      "      \"latex\": null,\n",
      "      \"type\": \"figure\"\n",
      "    },\n",
      "    \"FIGREF1\": {\n",
      "      \"text\": \"Biology 2003, 4:213\",\n",
      "      \"latex\": null,\n",
      "      \"type\": \"figure\"\n",
      "    }\n",
      "  },\n",
      "  \"back_matter\": []\n",
      "}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'Charles Darwin recognized that the distribution and form of parasites was evolutionarily significant. He noted, for instance, that \"... the Pediculi [lice] collected in different countries from the different races of man ... differ, not only in colour, but in the structure of their claws and limbs. In every case in which many specimens were obtained the differences were constant\" [1] . More recently, several research groups [2] [3] [4] [5] [6] [7] have found interesting correlations between the evolutionary relationships among various bacterial and viral strains hosted by humans and the pattern of migrations of modern humans throughout the world. A particularly interesting case is that of Helicobacter pylori, a Gram-negative bacterium associated with gastritis, peptic ulcers, and gastric cancer that may infect up to half of all humans [8] . The discovery that a bacterial infection could lead to what were considered chronic diseases [8] was a striking example of the fact that infectious diseases have not yet been conquered. The continuing acquired immune deficiency syndrome (AIDS) epidemic, outbreaks of Ebola in Central Africa, and the current spread of West Nile Virus in the United States and of severe acute respiratory syndrome (SARS) from Asia provide evidence of the pervasiveness and health consequences of infectious agents even in the age of vaccination and antimicrobial and antiviral therapies. Many infectious diseases are thought to have arisen concurrently with the development of agriculture and the rise of urban living. If, instead, many pathogens\\' relationships with humans are much older, it would not be surprising to find deeper evolutionary associations between humans and their microbial and viral invaders. The evolutionary history of H. pylori may provide an example of the coevolution of a bacterium and its only known host. The H. pylori genome is relatively small at 1.67 megabases, with a minimal complement of metabolic genes [9] . Variation between H. pylori isolates from different people or even from one person is large, leading to unique fingerprints for nearly every isolate so far typed. The coding genes are not very diverse, however: most of the variation occurs in the third base position within codons or through inversions or translocations, leaving the encoded amino-acid sequences relatively similar [2] . This amino-acid sequence conservation is fortunate for vaccine research, as one vaccine is likely to be effective on many strains. More interesting is the fact that H. pylori has an extremely high rate of recombination, higher in fact than that of any other organism characterized to date [3, 10] . Normally, such a high rate of recombination would make inferring an organism\\'s evolutionary history very difficult, as information about the origin of each mutation would be lost as it spreads throughout a population. But coupled with the mode of transmission of H. pylori, this extremely high rate of recombination may in fact make evolutionary inferences easier. Several studies strongly suggest that H. pylori is usually transmitted within families, generally from mother to child [11, 12] . Thus, the transmission of H. pylori in some ways mimics that of maternally transmitted mitochondrial DNA [13] . Because mitochondrial DNA is transmitted solely from one parent (the mother) and does not recombine, it has proved to be an ideal genetic system for inferring human evolutionary history (see below) [14, 15] . If H. pylori is indeed predominantly maternally transmitted, new strains will generally not infect a person during their lifetime; together with the high rate of recombination, this would mean that the mutations that accumulate within the population of bacteria in an individual\\'s stomach will be relatively homogeneous. This should result in a swarm of strains that are very closely related to each other, containing many of the mutations that have occurred in individual bacteria. Swarms found in different people will thus be more different from each other than if there was less recombination. Most infectious diseases spread quickly throughout the world and strains from different regions are relatively similar, but initial sampling of H. pylori from people from different regions of the world revealed fairly strong geographic partitioning into European and Asian H. pylori types [2] [3] [4] . Recently, Falush and colleagues [5] have examined this partitioning in greater detail. After sequencing eight genes -a total of 3,850 nucleotides -in 370 strains derived from 27 human populations, they found 1,418 polymorphic nucleotide positions. They then applied a new analytic tool, STRUCTURE [16] , that was developed to infer human genetic structure from multilocus genotype data. The program uses Bayesian methods to identify subgroups with distinctive allele frequencies, and clusters of the subgroups, even in the presence of recombination [16] . When this technique was applied to the H. pylori sequences, four main clusters were found -two from Africa and one each from Europe and Asia (Figure 1a ) [5] . Each cluster found by Falush and colleagues [5] could be divided into subgroups; for instance, the \\'Africa 1\\' cluster could be subdivided further into West and South African subclusters, and the East Asia cluster could be split into East Asian, Amerind, and Maori subclusters. The geographic partitioning within the 200 European strains was particularly complicated, presumably because numerous groups have swept back and forth across Europe over the past several millennia. European strains also occasionally appeared in the Americas, Australia, and among South Africans, presumably reflecting colonial conquest. The phylogenetic relationships of these clusters (Figure 1b) and their subdivisions [5] show a pattern similar to that The relationships between human populations, as calculated from H. pylori found in stomachs and from mitochondrial DNA data. (a) Relationships between modern subpopulations of H. pylori [5] . Each subpopulation is represented by a circle with a diameter proportional to the genetic diversity within it. The centres of the circles are joined by a phylogenetic tree showing the relationships between the four subpopulations. Bacteria in each subpopulation are found predominantly in people who originate from the regions shown. (b) A population-level phylogenetic tree of the H. pylori geographic subpopulations shown in (a). (c) A median-joining network of human populations derived from mitochondrial DNA [14] . Such a network shows alternative potential evolutionary relationships between clusters. Each circle represents a cluster of mitochondrial types with a diameter proportional to the frequency of that type within the subpopulations. All non-African populations are derived from one African lineage; the network of relationships within this lineage is magnified (top). (a,b) Adapted from [5] ; (c) adapted from [14] .  obtained using mitochondrial DNA variation (Figure 1c ) [14, 15] . The modern human gene pool, as inferred from mitochondrial DNA and corroborated by Y chromosome studies, is thought to have had an African origin approximately 150,000-200,000 years ago [14, 15, 17, 18] . The original human population then spread and diversified throughout Africa for nearly 100,000 years, before expanding into Western Asia and Europe and into Southern and Eastern Asia approximately 50,000-60,000 years ago, replacing the existing archaic populations of humans in these regions. Subsequent migrations spread into Australasia by 40,000 years ago, then to the Pacific Islands, and later into North America, approximately 15,000 years ago ( Figure 2 ) [14, 15, 17, 18] . The remarkable similarity between this view of human history and the results from studies of H. pylori have led Falush and colleagues [5] as well as others [2] to conclude that H. pylori evolution has followed the path of modern human expansion and migration. This work thus provides another type of data for analysis of human evolution and migration, independent of mitochondrial DNA and Y chromosomes, that will be valuable in further studies. Unfortunately, however, estimating the divergence dates from H. pylori is particularly difficult, owing to the extremely high rate of recombination [10] . Additional sampling and analytical techniques may be required to further test the migratory hypothesis. Other pathogens have also been proposed to follow evolutionary histories similar to their hosts. One of the most interesting examples, though with a non-human host, is that of aphids, a bacterium found within them, and two plasmids associated with the bacterium. Funk et al. [19] found that the inferred intraspecific phylogenies of these four genomes were completely congruent. Returning to human pathogens, the human polyomavirus JC virus (JCV) can be divided into genotypes that correspond to the major continental land masses [6] . Like H. pylori, JCV -which can cause progressive multifocal leukoencephalopathy (loss of myelination in the central nervous system) -is very widespread amongst humans as a result of familial transmission. A total of 12 known subtypes have been defined, with European, African, and Asian distributions [7] . Although direct inferences of an African origin for JCV are problematic because there is no suitable outgroup with which to root the phylogenetic tree, when an African origin is assumed, a reasonable evolutionary history can be hypothesized ( Figure 3 ) [7] . As with H. pylori, inferring molecular divergence dates is currently problematic for JCV. Further investigation into the evolutionary history of these human pathogens is therefore necessary. Elucidating the patterns of evolution of human pathogens may ultimately provide additional evidence not only about their history but also about human evolution and history. This will be especially true for pathogens such as H. pylori that have a predominantly mother-child mode of transmission, mimicking mitochondrial DNA evolution. H. pylori\\'s causative role in several chronic stomach conditions is just one of many current examples of the known or suspected  A map of the pattern of expansion and migration of modern humans throughout the world, derived from studies of mitochondrial DNA and Y chromosomes [14, 15, 17, 18] . Numbers indicate the approximate time (in years before the present) when modern humans first appeared in the indicated region. role of an infectious agent leading to chronic diseases. Bacteria are suspected to be involved in the development of arteriosclerosis, stroke, and Crohn\\'s disease, whereas viruses are known to lead to AIDS and the various forms of chronic hepatitis. Cervical cancer, hepatocellular carcinoma, Burkitt\\'s lymphoma, Kaposi\\'s sarcoma, and perhaps diabetes mellitus are also either known or suspected to be of viral origin. If some of the ubiquitous chronic diseases prove to be of bacterial or viral origin, worldwide surveys of these pathogens in human populations should be carried out immediately, so that knowledge about the evolution and diversity of these pathogens can be incorporated into the research programs designed to ameliorate the conditions that they cause. Relationships of human polyoma JC virus (JCV) subtypes found in humans from different parts of the world [7] . Letters refer to individual subtypes. (a) The hypothesized pattern of spread of JCV subtypes through the world (excluding the Americas); (b) an inferred phylogeny of JCV subtypes, assuming an African origin for the virus. Adapted from [7] .  '"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "row = df_sub.iloc[[1]]\n",
    "file_path = 'data/' + row.full_text_file[1] + '/' + row.full_text_file[1] + '/'\n",
    "\n",
    "if row.has_pdf_parse[1]:\n",
    "    file_path += 'pdf_json/' + row.sha[1] + '.json'\n",
    "else:\n",
    "    file_path += 'pmc_json/' + row.pmcid[1] + '.json'\n",
    "\n",
    "with open(file_path) as json_file:\n",
    "    data = json.load(json_file)\n",
    "    print(json.dumps(data, indent=2))\n",
    "    body_text = ''\n",
    "    for section in data['body_text']:\n",
    "        body_text += section['text'] + ' '\n",
    "body_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7 (tensorflow)",
   "language": "python",
   "name": "tensorflow"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
